packages/launcher/src/hooks/use-effect-router.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import useQuery from './use-query';
+import useRouter from './use-router';
+
+export type SearchProps = Partial<Record<string, unknown>>;
+
+export type UseEffectRouterReturns = {
+    redirectEffect(search?: SearchProps): void;
+    replaceEffect(search?: SearchProps): void;
+};
+
+const useEffectRouter = (): UseEffectRouterReturns => {
+    const { pathname, state } = useLocation();
+    const router = useRouter();
+    const query = useQuery();
+
+    const redirectEffect = useCallback(
+        search => {
+            router.redirect(
+                pathname,
+                {
+                    ...query,
+                    ...search,
+                },
+                state,
+            );
+        },
+        [query, pathname, router, state],
+    );
+
+    const replaceEffect = useCallback(
+        search => {
+            router.replace(
+                pathname,
+                {
+                    ...query,
+                    ...search,
+                },
+                state,
```

Comment:
no state

Reply:
why

Reply:
Because state is with the location,I think it should determine whether the user feels the current state is attached

Reply:
> Because state is with the location,I think it should determine whether the user feels the current state is attached

is lose？